MapboxNavigation/DataCache.swift

```diff
@@ -23,15 +23,7 @@ public class DataCache: NSObject, BimodalDataCache {
     public func store(_ data: Data, forKey key: String, toDisk: Bool, completion: CompletionHandler?) {
         storeDataInMemoryCache(data, forKey: key)
 
-        if toDisk == true {
-            fileCache.store(data, forKey: key, completion: completion)
-        } else {
-            if let completion = completion {
-                DispatchQueue.main.async {
-                    completion()
-                }
-            }
-        }
+        toDisk == true ? fileCache.store(data, forKey: key, completion: completion) : completion?()
```

Comment:
This was previously dispatched to the main queue. Are we sure we're on the main queue when we set the image?

Reply:
Yes. See [here](https://github.com/mapbox/mapbox-navigation-ios/pull/1272/files#diff-621dcac99b9b9fa73ac6931ca5a68ccdR22).

(PS: It's interesting to note that we don't actually _set_ the image through the callback. Although the callback does pass back an image, we actually implicitly rely on the image cache, because if you follow the callback chain, we essentially simply [call](https://github.com/mapbox/mapbox-navigation-ios/pull/1272/files#diff-af48d70309ee195e34cc5e6cc6b27caaR203) `attributedText()` again.)

Reply:
Jerrad and I paired on this -- the design decision that felt the most right was to require the caller to control the queue hopping, not burying it inside the lower-level objects. So a UI class needs to be explicit about making assignments on the main queue inside the completion handler, such as we're doing [here](https://github.com/mapbox/mapbox-navigation-ios/pull/1272/files#diff-621dcac99b9b9fa73ac6931ca5a68ccdR22)

We should make sure to call this out in the API documentation.